Repo: Corrected files to comply with .editorconfig rules.

EricDunsworth opened the pull request:
Brings all files in line with the rules set out in #8450's .editorconfig file (except comments).

Specific changes:
* Removed trailing spaces at the end of certain lines.
* Added missing final newlines.
* Removed sequences of multiple consecutive final newlines.
* Revised indentation styles to what they should be (tabs or 2 spaces).
* Removed BOM characters from the start of certain files as a part of indentation revisions.
* Added space character before "site/includes/sitemenu.hbs"'s reference to "sitemenureplace.hbs" and removed the former's leading space.
* Fixed indentation levels of certain JavaScript examples in "site/pages/docs/ref/geomap/geomap-*.hbs".
* Fixed indentation levels of certain section elements in "site/pages/docs/versions/dwnld-en.hbs".
* Reset indentation in "src/plugins/geomap/demo/sample.json".

nschonni commented:
Oops, maybe we should exclude these https://github.com/wet-boew/wet-boew/blob/master/.eslintignore since they are 3rd party JS

EricDunsworth commented:
I agree in principle (and don't mind revising this and #8450 accordingly). But I'm not sure how much value there'd be in exempting all of those files.

The only one that's currently being modified by this PR is [src/polyfills/slider/slider.js](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/slider/slider.js). Apart from that, all other files are already in line with the repo's EditorConfig settings.

Other things I noticed:
* [src/polyfills/events/mobile.js](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/events/mobile.js) and [src/plugins/deps/tableparser.js](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/deps/tableparser.js) look like legitimate third party scripts. But they're already in line with the repo's EditorConfig settings.
* [src/plugins/deps/tableparser.js](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/deps/tableparser.js) looks like it belongs to this repo. Doesn't look like it was sourced from an external project.
* [src/plugins/geomap/data/config-en.js](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/geomap/data/config-en.js) and [src/plugins/geomap/data/config-fr.js](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/geomap/data/config-fr.js) don't exist. Not even as untracked files in my local setup.

IMO only [src/polyfills/slider/slider.js](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/slider/slider.js) really needs to be exempted to avoid clashes. [src/polyfills/events/mobile.js](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/events/mobile.js) and [src/plugins/deps/tableparser.js](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/deps/tableparser.js) could also be worth exempting.

**PS:**
This isn't covered by .eslintignore, but I noticed that [dep/modernizr-custom.js](https://github.com/wet-boew/wet-boew/blob/dep/modernizr-custom.js) looks like it was generated from a tool on a third party website. This PR is currently adding a final newline to it. So I'll exempt it too.

@nschonni Thoughts?

EricDunsworth commented:
Just reverted this PR's changes to [src/polyfills/slider/slider.js](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/slider/slider.js) and [dep/modernizr-custom.js](https://github.com/wet-boew/wet-boew/blob/dep/modernizr-custom.js).

EricDunsworth commented on src/plugins/tabs/tabs-carousel-en.hbs:
```diff
@@ -301,27 +301,26 @@
+			<!--
 
-               -->
-            <h3>Carousel - Style 1 - Only one choice (slider controls and tranisitions are not applied)</h3>
```
Good catch! I probably inadvertently deleted it while messing around with lines 304-306. I'll fix it up soon.

EricDunsworth replied:
It's back in place now (check a few lines below).

EricDunsworth commented on src/plugins/geomap/demo/sample.json:
```diff
@@ -1,4109 +1,3325 @@
-      {
-      "rel": "self",          
+      "href": "http://geogratis.gc.ca/api/beta/en/nrcan-rncan/ess-sst"
+    }, {
```
Thanks for spotting this. I'll try to resolve it.

I was originally trying to use a mass find/replace to fix that file's indenting, but it didn't seem to be using a consistent indentation style... So I ended up using Dreamweaver's source indent tool to automatically re-indent it. I guess it deals with comma separations a bit differently.

EricDunsworth committed:
Repo: Corrected files to comply with .editorconfig rules.

Brings all files in line with the rules set out in #8450's .editorconfig file (except comments).

Specific changes:
* Removed trailing spaces at the end of certain lines.
* Added missing final newlines.
* Removed sequences of multiple consecutive final newlines.
* Revised indentation styles to what they should be (tabs or 2 spaces).
* Removed BOM characters from the start of certain files as a part of indentation revisions.
* Added space character before "site/includes/sitemenu.hbs"'s reference to "sitemenureplace.hbs" and removed the former's leading space.
* Fixed indentation levels of certain JavaScript examples in "site/pages/docs/ref/geomap/geomap-*.hbs".
* Fixed indentation levels of certain section elements in "site/pages/docs/versions/dwnld-en.hbs".
* Reset indentation in "src/plugins/geomap/demo/sample.json".

EricDunsworth commented:
@duboisp Just rebased, dumped in #8453's changes and re-ran ``grunt lintspaces`` - no issues were found. So IMO this PR can be merged in now.

Once that's done, next up would be to restart #8453's Travis-CI build and merge it in if it starts passing.
